content/news/XXXX-XX-XX-dvs-internals-pt1.md

```diff
@@ -0,0 +1,115 @@
+Instead, [digital signals](https://en.wikipedia.org/wiki/Digital_signal_(signal_processing)) works with measurements (called "samples") that indicate the value of the wave position at specific points in time.
+How often these measurements are taken is determined by the sample rate (or sampling rate).
+For example, Audio CDs use a sample rate of 44100 Hz (i.e. 44100 measurements per second, or 1 measurement every ~22.68 microseconds).
+
+To illustrate this, here are the individual samples in the timecode signal signified by little dots:
+
+![Individual samples in timecode signal]({static}/images/news/timecode-signal-samples.png)
+
+To detect zero crossings reliably, we cannot assume that such a sample measurement is happening at the exact time that the signal is at zero position.
+It might be the case the current sample is *after* the zero position and the previous sample was *before* the zero position.
+Hence, we also compare the previous sample value with the current sample value.
+If the current value is positive and the previous value was negative, or if the previous value was negative and the current value is positive, we crossed zero.
+
+
+## Refining Pitch Detection using Zero Crossings
+
+Now that we know how to detect zero crossings, we can use them for determining the pitch.
+
+If we look at the timecode signal, we can see that each cycle of the timecode signal has two zero crossings:
+At the start of the cycle, when the positive part of the wave start and in the middle of the cycle when the negative part of the wave starts.
+
+So the equivalent of checking for 1000 cycles per second is to check if there are 2000 zero crossings per second.
+Let's say the audio interface uses a sample rate of 44100 Hz, then there should be a be a zero crossing every 441000 / 2000 = 22.05 samples.
+
+We can now detect the pitch by comparing the expected number of samples between zero crossings with the actual number of samples between them.
+However, we need to use multiple sample distances and calculate the average to make this calculation more accurate, because in reality there are no fractions of samples, only full samples.
+
+
+## Detecting the Playback Direction
+
+The DVS system also needs to know if the record is playing forwards or backwards.
+
+To do that, we need to take a look at the right channel.
+It's basically the same signal as the left channel, but as we noticed before, it's shifted by a quarter cycle.
+This means that whenever there's a negative or positive amplitude peak in the left channel, the right channel is at zero.
+
+![Zero crossing on right channel means there is a peak on the left channel]({static}/images/news/timecode-signal-zero-crossing-right-channel.png)
+
+We can use this property to detect the playback direction.
+After the left channel crossed zero, we check if the left and right channel waves are both positive or both negative.
+If so, the record is playing forwards, otherwise it's playing backwards.
```

Comment:
```suggestion
After the left channel crossed zero, we check if the left and right channel waves are both positive.
If so, the record is playing forwards, otherwise it's playing backwards.
```

Reply:
This is wrong, the original sentence is correct.